### PR TITLE
DOCS-16161 100.7.1 Release Notes

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

100.7.1 Release notes and snooty version bump.
Also added "Amazon Linux 2023", "MaxOS 11 and later", and "RHEL 9" to the platform support matrix.

## STAGING

https://docs-mongodbcom-staging.corp.mongodb.com/database-tools/docsworker-xlarge/DOCS-16161/release-notes/database-tools-changelog/#100.7.1-changelog

https://docs-mongodbcom-staging.corp.mongodb.com/database-tools/docsworker-xlarge/DOCS-16161/installation/installation/#platform-support

## JIRA

https://jira.mongodb.org/browse/DOCS-16161

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=646fb13167a273b816f2fda3

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/includes/changelogs/releases/100.7.1.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.. _100.7.1-changelog:
+
+100.7.1 Changelog
+-----------------
+
+*Released 2023-05-24*
+
+This release fixes a few bugs and adds downloads for macOS 11 on ARM
+as well as RedHat Enterprise Linux 9 (x86 and ARM) and Amazon Linux
+2023 (x86 and ARM).
```
Comment (jeff-allen-mongo) [inline source/includes/changelogs/releases/100.7.1.rst:10]: The "as well as" and "and" in this sentence make this a little hard to parse in terms of which binary applies to which platform. What if we did something like:

This release fixes a few bugs and adds downloads for the following platforms:

- macOS 11 on ARM
- RedHat Enterprise Linux 9 (x86 and ARM)
- Amazon Linux 2023 (x86 and ARM)

Comment (jeff-allen-mongo) [inline source/includes/changelogs/releases/100.7.1.rst:10]: I'm not sure if we're tracking this in a separate ticket, but the [Platform Support matrix](https://docs-mongodbcom-staging.corp.mongodb.com/database-tools/docsworker-xlarge/DOCS-16161/installation/installation/#platform-support) doesn't mention RHEL 9. Sounds like it probably should.

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM with a formatting suggestion and a tangential comment.

**Comment by ianf-mongodb** [comment]
@FGasper Could you please take a look at this update when you get a moment?

**Review by FGasper — APPROVED** [review]
LGTM
